ir/src/main/scala/org/scalajs/core/ir/Tags.scala

```diff
@@ -90,6 +90,7 @@ private[ir] object Tags {
 
   // TODO Reorganize these when we can break binary compatibility
   final val TagJSSpread = TagModuleExportDef + 1
+  final val TagRestParamDef = TagJSSpread + 1
```

Comment:
Initially, I did it not to break deserialization of existing IR. It also has one advantage in general, though: 1 less byte for all parameters. I did not have a strong preference for either. Eventually the ease for deserialization made me choose this option. I can change it if you prefer.